public/modules/analysis/sps/sps.rb

```diff
@@ -1,86 +1,140 @@
 # -*- coding: utf-8 -*-
 
 require 'json'
-require_relative './collection'
-require_relative '../../../../lib/algorithms/sort_project'
-require_relative '../../../../lib/copy_peste/DbHdlr'
-require File.join CopyPeste::Require::Path.base, 'copy_peste/modules.rb'
+require 'parallel'
+require 'mongo'
+require File.join(CopyPeste::Require::Path.base, 'copy_peste/modules.rb')
+require File.join(CopyPeste::Require::Path.base, 'algorithms')
+require File.join(CopyPeste::Require::Path.analysis, 'fdf/config_handler/Ignored_class')
 
 class Sps < CopyPeste::Modules::Analysis
   attr_accessor :options
+  attr_accessor :show
 
 
-  # Initialize th Sps class
-  #
-  # @param [Array] Options for the Search project similarities (sps) module
-  def initialize(options)
-    @options = options
-    #@fichier = Collection.new("Fichier")
-    #@duplicate = Collection.new("Duplicate")
-    @tab_files = []
-    @sort_project = SortProject.new()
-    @mongo = DbHdlr.new()
-  end
+  def initialize
+    super()
 
+    @version = "1.0 - beta"
+    @authors = "CopyPeste Team"
+    @description = "Search average duplication between projects."
 
-  # Function called to start the Sps module
-  def start
-    get_file_from_db
-    sort_file
-    compare_projects
+    @options = {}
+    @results = {
+      module: "SPS",
+      options: "List of duplicated Project",
+      timestamp: Time.now.getutc,
+      data: []
+    }
+
+    @ignored_conf = Ignored_class.new()
   end
 
-  private
+  # Function called to start the Sps module
+  def analyse(result)
+    res = {}
+    if ((result_fdf = get_file_from_db) != nil)
+      result_fdf.results.each do |data|
+        if data._type == "ArrayResult"
```

Comment:
so basically you are doing
```ruby
some_array.each { |item|
  if item.class == SomeClass
    # ... something

    # else
    #   do nothing
  end
}
```
What about filtering first?

And more importantly, why are you checking that? You are not even supposed to know what you are receiving!